quantum/os_detection.h

```diff
@@ -27,6 +27,11 @@ typedef enum {
     OS_IOS,
 } os_variant_t;
 
+/* splits where the slave side needs to process for rgb/oled functionality
+ * define "SPLIT_DETECTED_OS_ENABLE" to enable  
+ */
+extern os_variant_t detected_os;
```

Comment:
This really shouldn't be exposed, TBH. 

Ideally should have a function to update it on the slave side. Eg: 
```c
#    if defined(SPLIT_KEYBOARD) && defined(SPLIT_DETECTED_OS_ENABLE)
void slave_update_detected_host_os(os_variant_t state) {
    detected_os = state;
}
#    endif
```